tests/ci.sh

```diff
@@ -111,62 +118,61 @@ RUST_BACKTRACE=1 cargo test --doc $FEATURES
+    echo "Execute all-providers tests"
+    RUST_BACKTRACE=1 cargo test $FEATURES all_providers
 else
-	# Per provider tests
-	echo "Execute normal tests"
-	RUST_BACKTRACE=1 cargo test $FEATURES normal_tests
-
-	echo "Execute persistent test, before the reload"
-	RUST_BACKTRACE=1 cargo test $FEATURES persistent_before
-
-	# Create a fake mapping file for the root application, the provider and a
-	# key name of "Test Key". It contains a valid PSA Key ID.
-	# It is tested in test "should_have_been_deleted".
-	# This test does not make sense for the TPM provider.
-	if [[ $1 = "mbed-crypto" ]]
-	then
-		echo "Create a fake mapping file for Mbed Provider"
-		mkdir -p mappings/cm9vdA==/1
-		printf '\xe0\x19\xb2\x5c' > mappings/cm9vdA==/1/VGVzdCBLZXk\=
-	elif [[ $1 = "pkcs11" ]]
-	then
-		echo "Create a fake mapping file for PKCS 11 Provider"
-		mkdir -p mappings/cm9vdA==/2
-		printf '\xe0\x19\xb2\x5c' > mappings/cm9vdA==/2/VGVzdCBLZXk\=
-	fi
-
-	echo "Trigger a configuration reload to load the new mappings"
-	kill -s SIGHUP $SERVER_PID
-	# Sleep time needed to make sure Parsec is ready before launching the tests.
-	sleep 5
-
-	echo "Execute persistent test, after the reload"
-	RUST_BACKTRACE=1 cargo test $FEATURES persistent_after
-
-	echo "Shutdown Parsec"
-	kill $SERVER_PID
-	# Sleep time needed to make sure Parsec is killed.
-	sleep 2
-
-	echo "Start Parsec for stress tests"
-	# Change the log level for the stress tests because logging is limited on the
-	# CI servers.
-	RUST_LOG=error RUST_BACKTRACE=1 cargo run $FEATURES -- --config $CONFIG_PATH &
-	SERVER_PID=$!
-	sleep 5
-
-	echo "Execute stress tests"
-	RUST_BACKTRACE=1 cargo test $FEATURES stress_test
+    # Per provider tests
+    echo "Execute normal tests"
+    RUST_BACKTRACE=1 cargo test $FEATURES normal_tests
+
+    echo "Execute persistent test, before the reload"
+    RUST_BACKTRACE=1 cargo test $FEATURES persistent_before
+
+    # Create a fake mapping file for the root application, the provider and a
+    # key name of "Test Key". It contains a valid PSA Key ID.
+    # It is tested in test "should_have_been_deleted".
+    # This test does not make sense for the TPM provider.
+    if [ "$1" = "mbed-crypto" ]
+    then
+        echo "Create a fake mapping file for Mbed Provider"
+        mkdir -p mappings/cm9vdA==/1
+        printf '\xe0\x19\xb2\x5c' > mappings/cm9vdA==/1/VGVzdCBLZXk\=
+    elif [ "$1" = "pkcs11" ]
+    then
+        echo "Create a fake mapping file for PKCS 11 Provider"
+        mkdir -p mappings/cm9vdA==/2
+        printf '\xe0\x19\xb2\x5c' > mappings/cm9vdA==/2/VGVzdCBLZXk\=
+    fi
+
+    echo "Trigger a configuration reload to load the new mappings"
+    kill -s SIGHUP $PARSEC_PID
+    # Sleep time needed to make sure Parsec is ready before launching the tests.
+    sleep 5
+
+    echo "Execute persistent test, after the reload"
+    RUST_BACKTRACE=1 cargo test $FEATURES persistent_after
+
+    echo "Shutdown Parsec"
+    kill $PARSEC_PID
```

Comment:
Don't know why I haven't asked until now, but why do we kill the server here? Would a reload suffice? @hug-dev

Reply:
The shutdown is needed because the logging level is changed for the stress tests. I guess we could modify the config file in place before that step to change the log level in the file and then reload it but I think that is simpler.

If we had the same logging level for stress tests and other tests, then we would not even have to reload it for the stress tests!